packages/react-components/react-positioning/src/createPositionManager.ts

```diff
@@ -0,0 +1,119 @@
+import { computePosition } from '@floating-ui/dom';
+import type { Middleware, Placement, Strategy } from '@floating-ui/dom';
+import type { PositionManager, TargetElement } from './types';
+import { debounce, writeArrowUpdates, writeContainerUpdates, getScrollParent } from './utils';
+
+interface PositionManagerOptions {
+  /**
+   * The positioned element
+   */
+  container: HTMLElement;
+  /**
+   * Element that the container will be anchored to
+   */
+  target: TargetElement;
+  /**
+   * Arrow that points from the container to the target
+   */
+  arrow: HTMLElement | null;
+  /**
+   * The value of the css `position` property
+   * @default absolute
+   */
+  strategy: Strategy;
+  /**
+   * [Floating UI middleware](https://floating-ui.com/docs/middleware)
+   */
+  middleware: Middleware[];
+  /**
+   * [Floating UI placement](https://floating-ui.com/docs/computePosition#placement)
+   */
+  placement?: Placement;
+}
+
+/**
+ * @returns manager that handles positioning out of the react lifecycle
+ */
+export function createPositionManager(options: PositionManagerOptions): PositionManager {
+  const { container, target, arrow, strategy, middleware, placement } = options;
+  let isFirstUpdate = true;
+  const scrollParents: Set<HTMLElement> = new Set<HTMLElement>();
+  const defaultView = container.ownerDocument.defaultView;
```

Comment:
```suggestion
  const targetWindow = container.ownerDocument.defaultView;
```

nit `defaultView` is confusing.